test/integration/basic/basic.go

```diff
@@ -179,5 +188,6 @@ func (r *BasicTestRunner) Run(ctx context.Context) {
 		r.T.Logf("Testing: %s\n", c.doc)
 		r.Setup(ctx, c.createOptsPublic, c.createOptsPrivate)
 		r.RunTests(ctx)
+		r.TearDown(ctx)
```

Comment:
the "tearDown" called outside this "for" is no longer needed, since now it is being called on every for loop iteration

Reply:
Two things. Yes that is true, but mainly because with the refactored implementation we can still
achieve the same thing, but creating a new instance of the `ClusterTestRunner` with a different
`NamespaceID`, like: `basic-scenario-1` or `basic-scenario-2` and therefore we can preserve the
original behavior.

In terms of performance, yes, this approach makes things a bit slower, but on the other side it
consumes less resources from the cluster at runtime.

If you think it might be better to run faster, I can apply a small change and keep it as before.

Reply:
> If you think it might be better to run faster, I can apply a small change and keep it as before.
If it is as easy as you say to implement. Please! :+1: